web/react/components/search_results.jsx

```diff
@@ -64,6 +65,11 @@ SearchItem = React.createClass({
                 dispatchError(err, "getPost");
             }
         );
+
+    var postChannel = ChannelStore.get(this.props.post.channel_id);
+    var teammate = postChannel.type === 'D' ? UserStore.getProfile(postChannel.name.substring(postChannel.name.indexOf('_')+2)).username : "";
```

Comment:
seems it's similar https://github.com/mattermost/platform/blob/master/web/react/components/sidebar.jsx#L268-L283